// @rainbow-me/rainbowkit@2.0.7/dist/wallets/walletConnectors/chunk-77F6UYEI.js
"use client";
import {
  getInjectedConnector,
  hasInjectedProvider
} from "./chunk-AWVQ5XXE.js";

// src/wallets/walletConnectors/rabbyWallet/rabbyWallet.ts
var rabbyWallet = () => ({
  id: "rabby",
  name: "Rabby Wallet",
  iconUrl: async () => (await import("./rabbyWallet-22VWIFCE.js")).default,
  rdns: "io.rabby",
  iconBackground: "#8697FF",
  installed: hasInjectedProvider({ flag: "isRabby" }),
  downloadUrls: {
    chrome: "https://chrome.google.com/webstore/detail/rabby-wallet/acmacodkjbdgmoleebolmdjonilkdbch",
    browserExtension: "https://rabby.io"
  },
  extension: {
    instructions: {
      learnMoreUrl: "https://rabby.io/",
      steps: [
        {
          description: "wallet_connectors.rabby.extension.step1.description",
          step: "install",
          title: "wallet_connectors.rabby.extension.step1.title"
        },
        {
          description: "wallet_connectors.rabby.extension.step2.description",
          step: "create",
          title: "wallet_connectors.rabby.extension.step2.title"
        },
        {
          description: "wallet_connectors.rabby.extension.step3.description",
          step: "refresh",
          title: "wallet_connectors.rabby.extension.step3.title"
        }
      ]
    }
  },
  createConnector: getInjectedConnector({ flag: "isRabby" })
});

export {
  rabbyWallet
};
