// @rainbow-me/rainbowkit@2.0.7/dist/wallets/walletConnectors/chunk-VGHLP2HT.js
"use client";
import {
  getInjectedConnector,
  hasInjectedProvider
} from "./chunk-AWVQ5XXE.js";

// src/wallets/walletConnectors/zealWallet/zealWallet.ts
var zealWallet = () => ({
  id: "zeal",
  name: "Zeal",
  iconUrl: async () => (await import("./zealWallet-K7JBLVKT.js")).default,
  iconBackground: "#fff0",
  installed: hasInjectedProvider({ flag: "isZeal" }),
  downloadUrls: {
    browserExtension: "https://zeal.app"
  },
  extension: {
    instructions: {
      learnMoreUrl: "https://zeal.app/",
      steps: [
        {
          description: "wallet_connectors.zeal.extension.step1.description",
          step: "install",
          title: "wallet_connectors.zeal.extension.step1.title"
        },
        {
          description: "wallet_connectors.zeal.extension.step2.description",
          step: "create",
          title: "wallet_connectors.zeal.extension.step2.title"
        },
        {
          description: "wallet_connectors.zeal.extension.step3.description",
          step: "refresh",
          title: "wallet_connectors.zeal.extension.step3.title"
        }
      ]
    }
  },
  createConnector: getInjectedConnector({ flag: "isZeal" })
});

export {
  zealWallet
};
